Source/Core/GoogleEarthEnterpriseMetadata.js

```diff
@@ -396,5 +471,8624 @@ define([
+            dbRootParser = createDbRootParser();
+        }
+
+        var url = that._url + 'dbRoot.v5?output=proto';
+        var proxy = that._proxy;
+        if (defined(proxy)) {
+            url = proxy.getURL(url);
+        }
+
+        var promise = loadArrayBuffer(url)
+            .then(function(buf) {
+                var encryptedDbRootProto = dbRootParser.EncryptedDbRootProto.decode(new Uint8Array(buf));
+
+                var byteArray = encryptedDbRootProto.encryptionData;
+                var offset = byteArray.byteOffset;
+                var end = offset + byteArray.byteLength;
+                var key = that.key = byteArray.buffer.slice(offset, end);
+
+                byteArray = encryptedDbRootProto.dbrootData;
+                offset = byteArray.byteOffset;
+                end = offset + byteArray.byteLength;
+                var dbRootCompressed = byteArray.buffer.slice(offset, end);
+                return taskProcessor.scheduleTask({
+                    buffer : dbRootCompressed,
+                    type : 'DbRoot',
+                    key: key
+                }, [dbRootCompressed]);
+            })
+            .then(function(result) {
+                var dbRoot = dbRootParser.DbRootProto.decode(new Uint8Array(result.buffer));
+                that.imageryPresent = defaultValue(dbRoot.imageryPresent, that.imageryPresent);
+                that.protoImagery = dbRoot.protoImagery;
+                that.terrainPresent = defaultValue(dbRoot.terrainPresent, that.terrainPresent);
+                if (defined(dbRoot.endSnippet) && defined(dbRoot.endSnippet.model)) {
+                    var model = dbRoot.endSnippet.model;
+                    that.negativeAltitudeExponentBias = defaultValue(model.negativeAltitudeExponentBias, that.negativeAltitudeExponentBias);
+                    that.negativeAltitudeThreshold = defaultValue(model.compressedNegativeAltitudeThreshold, that.negativeAltitudeThreshold);
+                }
+                if (defined(dbRoot.databaseVersion)) {
+                    that._quadPacketVersion = defaultValue(dbRoot.databaseVersion.quadtreeVersion, that._quadPacketVersion);
+                }
+                var providers = that.providers;
+                var providerInfo = defaultValue(dbRoot.providerInfo, []);
+                var count = providerInfo.length;
+                for(var i=0;i<count;++i) {
+                    var provider = providerInfo[i];
+                    var copyrightString = provider.copyrightString;
+                    if (defined(copyrightString)) {
+                        providers[provider.providerId] = new Credit(copyrightString.value);
+                    }
+                }
+            })
+            .otherwise(function() {
+                // Just eat the error and use the default values.
+                console.log('Failed to retrieve ' + url + '. Using defaults.');
+                that.key = stringToBuffer(defaultKey);
```

Comment:
Why don't we just initialize `defaultKey` to the result of `stringToBuffer` instead of calling it every time?